### ISSUE TITLE
[BUG] (Compile Error LPC1768)

### ISSUE THREAD

**Issue by borland1** [MarlinFirmware/Marlin#16142]
### Bug Description

Compile 'Fails' if display MKS_12864OLED_SSD1306 is enabled with processor LPC1768,

But compile 'Succeeds' in PlatformIO (or Arduino IDE) with:  
default_envs = megaatmega2560
& 8-bit motherboard: MKS_GEN_L

### My Configurations
Marlin release version: 2.0.x
PlatformIO:   default_envs = LPC1768

### Steps to Reproduce

Compile 'Fails' with errors when enabling the following two lines
in file Configuration.h:

line 133, #define MOTHERBOARD BOARD_MKS_SGEN_L
line 2004,  #define MKS_12864OLED_SSD1306;

#### Additional Information

[ConfigFiles.zip](https://github.com/MarlinFirmware/Marlin/files/3935666/ConfigFiles.zip)

**Comment by thisiskeithb** [comment]
It's because the SGEN-L isn't setup for `MKS_12864OLED_SSD1306`. Looking at the [pins file](https://github.com/MarlinFirmware/Marlin/blob/39fcb5b5f891cea26b1e8ab0d4b749565e830582/Marlin/src/pins/lpc1768/pins_MKS_SGEN_L.h), only the following LCDs/controllers are setup:

- `CR10_STOCKDISPLAY`
- `FYSETC_MINI_12864`
- `FYSETC_MINI_12864_1_2`
- `FYSETC_MINI_12864_2_0`
- `FYSETC_MINI_12864_2_1`
- `MKS_MINI_12864`
- `ULTIPANEL`

~If you have the `MKS_12864OLED_SSD1306` pinout, we can use that to map the pins.~

Edit: Can you try the attached firmware to see if everything works:
[16142.zip](https://github.com/MarlinFirmware/Marlin/files/3935776/16142.zip)

I believe the correct pinout (using SD on the controller) is:
```cpp
#define BEEPER_PIN   P1_31
#define BTN_ENC      P1_30
#define DOGLCD_MOSI  P0_18
#define DOGLCD_CS    P0_16
#define DOGLCD_SCK   P0_15
#define LCD_PINS_DC  P0_17
#define DOGLCD_A0    LCD_PINS_DC
#define LCD_PINS_RS  P1_00
#define LCD_PINS_D7  P1_22
#define BTN_EN1      P3_25
#define BTN_EN2      P3_26
#define SD_DETECT_PIN P0_27
#define KILL_PIN     -1 // NC
```

**Comment by borland1** [comment]
I can confirm that the MKS_12864OLED_SSD1306 display does work with the provided Firmware.bin file.
I also confirmed that SD-card data is read from this displays SD-card slot.  So, apparently you have the pin assignments correct for both ports EXP1 and EXP2.  

Thank you for the effort.

**Comment by thisiskeithb** [comment]
Thanks for confirming the pinout is correct!

~~I’ll put in a PR later to get this screen/controller support added to the SGen-L.~~
[Link to PR](https://github.com/MarlinFirmware/Marlin/pull/16188).

**Comment by thisiskeithb** [comment]
@borland1: The fix was merged into the latest `bugfix-2.0.x` code, so can you download a fresh copy & make sure everything works as intended?

**Comment by boelle** [comment]
since a fix has been merged i will close this one

we can always reopen
